ingestion/src/metadata/profiler/metrics/static/distinct_count.py

```diff
@@ -41,15 +42,30 @@ def metric_type(self):
 
     @_label
     def fn(self):
+        """
+        Distinct Count metric for Sqlalchemy connectors
+        """
         return func.count(distinct(CountFn(column(self.col.name, self.col.type))))
 
     def df_fn(self, dfs=None):
+        """
+        Distinct Count metric for Datalake
+        """
         from collections import Counter  # pylint: disable=import-outside-toplevel
 
         try:
             counter = Counter()
             for df in dfs:
-                counter.update(df[self.col.name].dropna().to_list())
+                col_type = df[self.col.name].dtypes.name
+                if (
+                    col_type == "object"
+                    and self.col.type in [DataType.ARRAY, DataType.JSON]
+                    and any(df[self.col.name].values)
+                ):
+                    df_col_value = df[self.col.name].dropna().to_list()[0]
```

Comment:
can you give more details regarding this?

Reply:
@ayush-shah on top of the above, do we have unit tests for the pandas interface similar to the ones we have for SQA? https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/tests/unit/profiler/sqlalchemy/test_metrics.py

Reply:
yes @pmbrull we do have test_metrics for pandas, i will add better data to cover more scenarios